zipline/data/bundles/alpha_vantage_api.py

```diff
@@ -0,0 +1,323 @@
+
+                # dont know better way to convert _csv.reader to list or DataFrame
+                table = []
+                for line in data_slice:
+                  table.append(line)
+                
+                # strip header-row from csv
+                table = table[1:]
+
+                data = data + table
+        
+
+        df = pd.DataFrame(data, columns=['time', 'open', 'high', 'low', 'close', 'volume'])
+
+        df.rename(columns={'time': 'date'}, inplace=True)
+        df.index = pd.to_datetime(df['date'])
+        df.index = df.index.tz_localize('UTC') 
+        df.drop(columns=['date'], inplace=True)
+        
+    else:
+        data = av_api_wrapper(symbol, interval)
+
+        df = pd.DataFrame.from_dict(data, orient='index')
+        df.index = pd.to_datetime(df.index).tz_localize('UTC')
+        
+        df.rename(columns={
+          '1. open': 'open', 
+          '2. high': 'high', 
+          '3. low': 'low', 
+          '4. close': 'close', 
+          '5. volume': 'volume',
+          '5. adjusted close': 'adj_close',
+          '6. volume': 'volume',
+          '7. dividend amount': 'dividend', 
+          '8. split coefficient': 'split'
+          }, inplace=True)
+
+        #if '5. adjusted close' in df.columns:
+        #    df.drop('5. adjusted close', inplace=True, axis=1)
+    
+    df.sort_index(inplace=True)
+  
+    # data comes as strings
+    df['open'] = pd.to_numeric(df['open'], downcast='float') 
+    df['high'] = pd.to_numeric(df['high'], downcast='float') 
+    df['low'] = pd.to_numeric(df['low'], downcast='float') 
+    df['close'] = pd.to_numeric(df['close'], downcast='float') 
+    df['volume'] = pd.to_numeric(df['volume'], downcast='unsigned') 
+
+    if 'adj_close' in df.columns:
+        df['adj_close'] = pd.to_numeric(df['adj_close'], downcast='float') 
+
+    if 'dividend' in df.columns:
+        df['dividend'] = pd.to_numeric(df['dividend'], downcast='float') 
+
+    if 'split' in df.columns:
+        df['split'] = pd.to_numeric(df['split'], downcast='float') 
+
+    df.to_csv('/tmp/' + symbol + '_' + interval + '.csv')
+
+    return df
+         
+
+# collect all days where there were splits and calculate split-ratio
+# by 1 / split-factor. save them together with effective-date.
+def calc_split(sid, df):
+    tmp = 1. / df[df['split'] != 1.0]['split']
+    split = pd.DataFrame(data=tmp.index.tolist(),
+                      columns=['effective_date'])
+    split['ratio'] = tmp.tolist()
+    split['sid'] = np.int(sid)
+
+    split['effective_date'] = split['effective_date'].apply(lambda x: x.timestamp())
+
+    return split
+
+
+# collect all dividends and the dates when they were issued,
+# fill stuff we don't know with empty-values
+def calc_dividend(sid, df):
+    tmp = df[df['dividend'] != 0.0]['dividend']
+    div = pd.DataFrame(data=tmp.index.tolist(), columns=['ex_date'])
+    div['record_date'] = pd.NaT
+    div['declared_date'] = pd.NaT
```

Comment:
you have to document why you are filling these with NaT:

![image](https://user-images.githubusercontent.com/7924802/104704173-fcfde600-5720-11eb-8187-5b9378db3286.png)

Reply:
logic was taken out of the csvdir-bundle